scan-translations.ts

```diff
@@ -0,0 +1,465 @@
+    let keys: string[] = [];
+
+    for (const key in obj) {
+        const fullKey = prefix ? `${prefix}.${key}` : key;
+        const value = obj[key];
+
+        if (typeof value === "object" && value !== null && !Array.isArray(value)) {
+            keys = keys.concat(flattenKeys(value as TranslationKeys, fullKey));
+        } else {
+            keys.push(fullKey);
+        }
+    }
+
+    return keys;
+}
+
+/**
+ * Remove comments from content to avoid detecting translation keys in comments
+ * This function carefully preserves // in URLs and strings while removing actual comments
+ */
+export function stripComments(content: string): string {
+    // Remove multi-line comments (/* ... */) first
+    let result = content.replaceAll(/\/\*[\s\S]*?\*\//g, '');
+
+    // Remove single-line comments, but be careful not to match:
+    // - https:// or http:// in URLs
+    // - // inside strings
+    // This regex matches // that is NOT preceded by : (to avoid https://)
+    // and removes everything after it until end of line
+    result = result.replaceAll(/(?<!:)\/\/.*$/gm, '');
+
+    return result;
+}
+
+/**
+ * Extract translation keys from a file's content
+ */
+export function extractKeysFromContent(content: string): string[] {
+    const keys: string[] = [];
+
+    // Strip comments first to avoid detecting keys in comments
+    const contentWithoutComments = stripComments(content);
+
+    for (const pattern of TRANSLATION_PATTERNS) {
+        let match: RegExpExecArray | null = null;
+        // Reset lastIndex for global regex
+        pattern.lastIndex = 0;
+
+        while ((match = pattern.exec(contentWithoutComments)) !== null) {
+            const key = match[1];
+            // Skip dynamic keys (containing variables like ${}, {{}} etc.)
+            if (!key.includes("${") && !key.includes("{{") && !key.includes("}")) {
+                keys.push(key);
+            }
+        }
+    }
+
+    return keys;
+}
+
+/**
+ * Scan source files for translation keys
+ */
+async function scanSourceFiles(): Promise<Set<string>> {
+    console.log("🔍 Scanning source files for translation keys...\n");
+
+    const usedKeys = new Set<string>();
+
+    // Find all TypeScript and TypeScript React files
+    const files = await glob("**/*.{ts,tsx}", {
+        cwd: WEB_APP_DIR,
+        ignore: EXCLUDE_DIRS,
+        absolute: true,
+    });
+
+    console.log(`Found ${files.length} files to scan\n`);
+
+    // Read all files in parallel for better performance
+    const fileContents = await Promise.all(
+        files.map(async (file) => {
+            try {
+                const content = await fs.promises.readFile(file, "utf-8");
+                return { file, content };
+            } catch (error) {
+                console.error(`❌ Error: Could not read file ${file}: ${error}`);
+                return null;
+            }
+        })
+    );
+
+    // Process the contents
+    for (const result of fileContents) {
```

Comment:
we should add a check for the truthyness of `fileContents` because it can be null